src/components/Confirm/Confirm.tsx

```diff
@@ -197,24 +269,23 @@ class Confirm extends Component {
       imageQualityRetries,
       isDecoupledFromAPI,
     } = this.props
+
+    if (!token) {
+      throw new Error('token not provided')
```

Comment:
Do they need the token when `isDecoupledFromAPI === true` ?

Reply:
`onSubmitCallback` called when `isDecoupledFromAPI === true` is using the token.